gemrb/core/EffectQueue.cpp

```diff
@@ -1585,13 +1585,14 @@ void EffectQueue::RemoveAllNonPermanentEffects()
 //FIXME: probably this isn't perfect
 void EffectQueue::RemoveLevelEffects(ieDword level, ieDword Flags, ieDword match)
 {
+	bool found_one = false;
 	ResRef removed;
 	for (auto& fx : effects) {
 		if (fx.Power > level) {
 			continue;
 		}
 
-		if (removed != fx.SourceRef) {
+		if (found_one && removed != fx.SourceRef) {
```

Comment:
I think it would be cleaner to just check `!removed.IsEmpty()` instead of adding another variable.

Reply:
I agree.  I think when I wrote this I was still unclear how your custom string types worked.  Some of this code is several years old, actually, I just never submitted it for some reason.  Anyway, it's the last commit, so I'll just amend it.